pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -0,0 +1,153 @@
+}
+
+func (a *v1beta1PolicyApplier) JwtFilter(isXDSMarshalingToAnyEnabled bool) *http_conn.HttpFilter {
+	var filterConfigProto *envoy_jwt.JwtAuthentication
+	filterConfigProto = convertToEnvoyJwtConfig(a.processedJwtRules)
+
+	if filterConfigProto == nil {
+		return nil
+	}
+	out := &http_conn.HttpFilter{
+		Name: authn_model.EnvoyJwtFilterName,
+	}
+	if isXDSMarshalingToAnyEnabled {
+		out.ConfigType = &http_conn.HttpFilter_TypedConfig{TypedConfig: util.MessageToAny(filterConfigProto)}
+	} else {
+		out.ConfigType = &http_conn.HttpFilter_Config{Config: util.MessageToStruct(filterConfigProto)}
+	}
+	return out
+}
+
+func (a *v1beta1PolicyApplier) AuthNFilter(proxyType model.NodeType, isXDSMarshalingToAnyEnabled bool) *http_conn.HttpFilter {
+	// TODO(diemtvu) implement this.
+	log.Errorf("AuthNFilter(%v, %v) is not yet implemented", proxyType, isXDSMarshalingToAnyEnabled)
+	return nil
+}
+
+func (a *v1beta1PolicyApplier) InboundFilterChain(sdsUdsPath string, meta *model.NodeMetadata) []plugin.FilterChain {
+	// TODO(diemtvu) implement this.
+	log.Errorf("InboundFilterChain is not yet implemented")
+	return nil
+}
+
+// NewPolicyApplier returns new applier for v1beta1 authentication policies.
+func NewPolicyApplier(jwtPolicies []*model.Config) authn.PolicyApplier {
+	processedJwtRules := []*v1beta1.JWT{}
+
+	// TODO(diemtvu) should we need to deduplicate JWT with the same issuer.
+	for idx := range jwtPolicies {
+		spec := jwtPolicies[idx].Spec.(*v1beta1.RequestAuthentication)
+		for _, rule := range spec.JwtRules {
+			processedJwtRules = append(processedJwtRules, rule)
+		}
+	}
+
+	return &v1beta1PolicyApplier{
+		jwtPolicies:       jwtPolicies,
+		processedJwtRules: processedJwtRules,
+	}
+}
+
+func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWT) *envoy_jwt.JwtAuthentication {
+	if len(jwtRules) == 0 {
+		return nil
+	}
+	providers := map[string]*envoy_jwt.JwtProvider{}
+	for i, jwtRule := range jwtRules {
+		// TODO(diemtvu): set forward based on input spec after https://github.com/istio/api/pull/1172
+		provider := &envoy_jwt.JwtProvider{
+			Issuer:            jwtRule.Issuer,
+			Audiences:         jwtRule.Audiences,
+			Forward:           false,
+			PayloadInMetadata: jwtRule.Issuer,
+		}
+
+		for _, location := range jwtRule.FromHeaders {
+			provider.FromHeaders = append(provider.FromHeaders, &envoy_jwt.JwtHeader{
+				Name:        location.Name,
+				ValuePrefix: location.Prefix,
+			})
+		}
+		provider.FromParams = jwtRule.FromParams
+
+		jwtPubKey := jwtRule.Jwks
+		if jwtPubKey == "" {
+			var err error
+			jwtPubKey, err = model.JwtKeyResolver.GetPublicKey(jwtRule.JwksUri)
+			if err != nil {
+				log.Errorf("Failed to fetch jwt public key from %q: %s", jwtRule.JwksUri, err)
+			}
+		}
+		provider.JwksSourceSpecifier = &envoy_jwt.JwtProvider_LocalJwks{
+			LocalJwks: &core.DataSource{
+				Specifier: &core.DataSource_InlineString{
+					InlineString: jwtPubKey,
+				},
+			},
+		}
+
+		name := fmt.Sprintf("origins-%d", i)
```

Comment:
is `origins` a magic reserved prefix in envoy jwt filter?

Reply:
No. Just need to be unique.